packages/core/src/components/ModalNew/hooks/usePortalTarget/usePortalTarget.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import { PortalTarget } from "./usePortalTarget.types";
+
+const usePortalTarget = (portalTarget?: PortalTarget): Element | DocumentFragment => {
+  const [resolvedTarget, setResolvedTarget] = useState<Element | DocumentFragment>(document.body);
+
+  useEffect(() => {
+    const resolveTarget = (): Element | DocumentFragment => {
+      if (!portalTarget) {
+        return document.body;
+      }
+
+      if (typeof portalTarget === "string") {
+        const target = document.querySelector(portalTarget);
+        if (!target) {
+          console.warn(`Target with selector "${portalTarget}" not found. Falling back to document.body.`);
+          return document.body;
+        }
+        return target;
+      }
+
+      if (portalTarget instanceof Element || portalTarget instanceof DocumentFragment) {
+        return portalTarget;
+      }
+
+      if (portalTarget.current instanceof Element) {
+        return portalTarget.current;
+      }
+
+      console.warn("Invalid portalTarget provided. Falling back to document.body.");
```

Comment:
Do we need to print it in prod?

Reply:
I think it's better than printing it in prod, no? We did something similar in deprecateMessage